### DISCUSSION TITLE
How to switch conditionally a task in a step?

### DISCUSSION THREAD

**Discussion by burningwave** [community/community#26976] [Actions]
<p>I have created a workflow that contains this 2 jobs:</p>
<pre><code>build-and-test-with-Java-8-and-later:
    name: Build -&gt; Test (JVM ${{ matrix.java }}, ${{ matrix.architecture }}, ${{ matrix.os }})
    strategy:
      fail-fast: true
      matrix:
        os: [ubuntu-latest, macOS-latest, windows-latest]
        java: [8, 9, 10, 11, 12, 13, 14]
        architecture: [x32, x64]
    runs-on: ${{ matrix.os }}
    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK ${{ matrix.java }}
        uses: actions/setup-java@v1
        with:
          java-version: ${{ matrix.java }}
          architecture: ${{ matrix.architecture }}
      - name: Build and test
        run: mvn -B clean test -Dproject_jdk_version=${{ matrix.java }} -DskipTests=false --file pom.xml

  build-and-test-with-Java-15-ea:
    name: Build -&gt; Test (JVM ${{ matrix.java }}, ${{ matrix.architecture }}, ${{ matrix.os }})
    strategy:
      fail-fast: true
      matrix:
        os: [ubuntu-latest, macOS-latest, windows-latest]
        java: [15-ea]
        architecture: [x32, x64]
    runs-on: ${{ matrix.os }}
    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK ${{ matrix.java }}
        uses: actions/setup-java@v1
        with:
          java-version: ${{ matrix.java }}
          architecture: ${{ matrix.architecture }}
      - name: Build and test
        run: mvn -B clean test -Dproject_jdk_version=15 -DskipTests=false --file pom.xml
</code></pre>
<p>… And I would like to merge this 2 jobs in a unique job that switch the run element when iterated ${{ matrix.java }} variable contains the ‘-ea’ suffix. in particular I would like to create something like this:</p>
<pre><code>build-and-test-with-Java-8-and-later:
    name: Build -&gt; Test (JVM ${{ matrix.java }}, ${{ matrix.architecture }}, ${{ matrix.os }})
    strategy:
      fail-fast: true
      max-parallel: 10
      matrix:
        os: [ubuntu-latest, macOS-latest, windows-latest]
        java: [8, 9, 10, 11, 12, 13, 14, 15-ea]
        architecture: [x32, x64]
    runs-on: ${{ matrix.os }}
    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK ${{ matrix.java }}
        uses: actions/setup-java@v1
        with:
          java-version: if${{ matrix.java }}
          architecture: ${{ matrix.architecture }}
      - name: Build and test
        if: ${{ matrix.java }} != '15-ea'
        run: mvn -B clean test -Dproject_jdk_version=${{ matrix.java }} -DskipTests=false --file pom.xml
      - name: Build and test
        if: ${{ matrix.java }} == '15-ea'
        run: mvn -B clean test -Dproject_jdk_version=15 -DskipTests=false --file pom.xml
</code></pre>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/burningwave">@burningwave</a> ,</p>
<p>In if expression, please directly use matrix.java without ‘${{}}’. Code sample as below:</p>
<pre><code>build-and-test-with-Java-8-and-later:
    name: Build -&gt; Test (JVM ${{ matrix.java }}, ${{ matrix.architecture }}, ${{ matrix.os }})
    strategy:
      fail-fast: false
      max-parallel: 10
      matrix:
        os: [ubuntu-latest, macOS-latest, windows-latest]
        java: [8, 9, 10, 11, 12, 13, 14, 15-ea]
        architecture: [x32, x64]
    runs-on: ${{ matrix.os }}
    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK ${{ matrix.java }}
        uses: actions/setup-java@v1
        with:
          java-version: ${{ matrix.java }}
          architecture: ${{ matrix.architecture }}
      - name: Build and test
        if:&lt;font color="#FF0000"&gt;matrix.java != '15-ea'&lt;/font&gt; run: echo ${{ matrix.java }}
      - name: Build and test
        if:&lt;font color="#FF0000"&gt;matrix.java == '15-ea'&lt;/font&gt; run: echo ${{ matrix.java }}
</code></pre>
<p>Thanks.</p>

**Comment by burningwave** [comment]
<p>Thans a lot: I solved this is the final job:</p>
<pre><code>build-and-test-with-Java-8-and-later:
    name: Build -&gt; Test (JVM ${{ matrix.java }}, ${{ matrix.architecture }}, ${{ matrix.os }})
    strategy:
      fail-fast: true
      max-parallel: 10
      matrix:
        os: [ubuntu-latest, macOS-latest, windows-latest]
        java: [8, 9, 10, 11, 12, 13, 14, 15-ea]
        architecture: [x32, x64]
    runs-on: ${{ matrix.os }}
    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK ${{ matrix.java }}
        uses: actions/setup-java@v1
        with:
          java-version: ${{ matrix.java }}
          architecture: ${{ matrix.architecture }}
      - if: endsWith(matrix.java, '-ea') != true
        name: Build and test for final JDK version
        run: mvn -B clean test -Dproject_jdk_version=${{ matrix.java }} -DskipTests=false --file pom.xml
      - if: endsWith(matrix.java, '-ea') == true
        name: Build and test for Early Access JDK version
        run: mvn -B clean test -Dproject_jdk_version=15 -DskipTests=false --file pom.xml
</code></pre>
<p>… and do you know how to replace a String for writing something like this in the last row?:</p>
<pre><code>run: mvn -B clean test -Dproject_jdk_version=${{matrix.java.replace('-ea', '')}} -DskipTests=false --file pom.xml
</code></pre>
